docs/advanced/porting_guide.md

```diff
@@ -37,42 +34,43 @@ Add the target description to ```mbed-os\targets\targets.json```:
 
-Dynamic vector relocation also requires some extensions to CMSIS-CORE. ```mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis.h``` is the entry point to the target for mbed OS software. This will include the vector relocation additions and device-specific headers that include CMSIS-CORE. A relocation routine is needed in ```mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis_nvic.c and .h```
+CMSIS-RTOS needs a few macros to initialize the SYS_TICK. These macros are usually configured in`mbed-os\targets\TARGET_VENDOR\mbed-rtx.h`. 
 
-Now verify your target is identified by using ```mbed compile -m MCU_NAME -t <toolchain>```. This compiles.
+Now verify that your target compiles by using `mbed compile -m MCU_NAME -t <toolchain>`.
 
-The next step is to enable the test harness dependencies. To run the test suite, there is a minimum requirement of GPIO, microsecond ticker and serial implementation, all explained below.
+The next step to port a target is to enable the test harness dependencies. To run the test suite, your target must support GPIO, microsecond ticker and serial.
 
 ### GPIO
 
-The ```gpio_s``` is for referencing memory-mapped GPIO registers and passing related pin information or other IO operation data that the HAL needs. These structures are defined in ```objects.h```. The required implementation definition is in ```mbed-os\hal\gpio_api.h```.
-
-### Serial
-
-The ```serial_s``` is for referencing memory-mapped serial registers and passing related pin and peripheral operation information data that the HAL needs. These structures are defined in ```objects.h```. The required implementation definition is in ```mbed-os\hal\serial_api.h```.
+Implement the api declared in `mbed-os/hal/gpio_api.h`. You must define the struct `gpio_t`. This struct is commonly defined in an `objects.h` file within the `mbed-os/targets/TARGET_VENDOR/`, `mbed-os/targets/TARGET_VENDOR/TARGET_MCU_FAMILY` or `mbed-os/targets/TARGET_VENDOR/TARGET_MCU_FAMILY/TARGET_MCUNAME` directories.
 
 ### Microsecond ticker
 
-The microsecond ticker is a system resource that many APIs and other timing utilities use. It needs a one microsecond resolution and should be implemented using a free-running hardware counter or timer with match register. The required implementation definition is in ```mbed-os\hal\us_ticker_api.h```.
+The microsecond ticker is a system resource that many APIs use. It needs a one microsecond resolution and should be implemented using a free-running hardware counter or timer with match register. Implement the api declared in `mbed-os\hal\us_ticker_api.h`.
```

Comment:
- Please change "resolution and should be implemented using" to "resolution. Implement it by using".
- Please capitalize "API".